src/Wave.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef GAZEBO_WAVE_PLUGIN_HPP
+#define GAZEBO_WAVE_PLUGIN_HPP
+
+#include <gazebo/common/common.hh>
+#include <gazebo/physics/physics.hh>
+#include <gazebo/transport/transport.hh>
+#include <ignition/math/Vector3.hh>
+#include "base/Time.hpp"
+
+namespace gazebo_usv
+{
+    class Wave
+    {
+        typedef gazebo::physics::ModelPtr ModelPtr;
+        typedef gazebo::physics::LinkPtr LinkPtr;
+        typedef gazebo::transport::NodePtr NodePtr;
+        typedef gazebo::transport::SubscriberPtr SubscriberPtr;
+
+    public:
+        // Parameters used to calculate the Wave effects 
+        struct EffectParameters
+        {
+            double torque_constant;
+        };
+        // Wave force and torque
+        struct Effects
+        {
+            ignition::math::Vector3d force;
+            ignition::math::Vector3d torque;
+        };
+        //Wave phases to be defined
+        double phase_x;
+        double phase_y;
+        double phase_z;
```

Comment:
Since the phases are public just to make testing it easier, I think you should still name them following the private member variables guidelines (e.g. m_phase_x)